server/src/main/java/com/kritsit/casetracker/server/domain/services/Login.java

```diff
@@ -1,18 +1,26 @@
 package com.kritsit.casetracker.server.domain.services;
 
-import java.util.Arrays;
+import com.kritsit.casetracker.server.datalayer.IUserRepository;
+import com.kritsit.casetracker.server.datalayer.RowToModelParseException;
+import com.kritsit.casetracker.server.domain.model.Staff;
 
 public class Login implements ILoginService {
-    private IPersistenceService persistence;
+    private IUserRepository repository;
 
-    public Login(IPersistenceService persistence) {
-        this.persistence = persistence;
+    public Login(IUserRepository repository) {
+        this.repository = repository;
     }
 
-    public boolean login(String username, int passwordHash) {
-        long passwordSaltedHash = persistence.getPasswordSaltedHash(username);
-        long salt = persistence.getSalt(username);
+    public Staff login(String username, int passwordHash) throws RowToModelParseException {
+        long passwordSaltedHash = repository.getPasswordSaltedHash(username);
+        long salt = repository.getSalt(username);
         long testPasswordSaltedHash = salt + passwordHash;
-        return !(passwordSaltedHash == -1 || testPasswordSaltedHash != passwordSaltedHash);
+        boolean isAuthenticated = !(passwordSaltedHash == -1 || testPasswordSaltedHash != passwordSaltedHash);
+        
+        if(!isAuthenticated){
+        	//throw custom exception?
```

Comment:
I thouht Login needed to return a Staff object. I like the idea of using HTTP status codes :+1: